nautobot/extras/datasources/git.py

```diff
@@ -554,6 +571,158 @@ def delete_git_config_contexts(repository_record, job_result, preserve=(), prese
                 )
 
 
+#
+# Config Context Schemas
+#
+
+
+def refresh_git_config_context_schemas(repository_record, job_result, delete=False):
+    """Callback function for GitRepository updates - refresh all ConfigContextSchema records managed by this repository."""
+    if "extras.configcontextschema" in repository_record.provided_contents and not delete:
+        update_git_config_context_schemas(repository_record, job_result)
+    else:
+        delete_git_config_context_schemas(repository_record, job_result)
+
+
+def update_git_config_context_schemas(repository_record, job_result):
+    """Refresh any config context schemas provided by this Git repository."""
+    config_context_schema_path = os.path.join(repository_record.filesystem_path, "config_context_schemas")
+    if not os.path.isdir(config_context_schema_path):
+        return
+
+    managed_config_context_schemas = set()
+
+    for file_name in os.listdir(config_context_schema_path):
+        if not os.path.isfile(os.path.join(config_context_schema_path, file_name)):
+            continue
+        job_result.log(
+            f"Loading config context schema from `{file_name}`",
+            grouping="config contexts",
+            logger=logger,
+        )
+        try:
+            with open(os.path.join(config_context_schema_path, file_name), "r") as fd:
+                # The data file can be either JSON or YAML; since YAML is a superset of JSON, we can load it regardless
+                try:
+                    context_schema_data = yaml.safe_load(fd)
+                except Exception as exc:
+                    raise RuntimeError(f"Error in loading config context schema data from `{file_name}`: {exc}")
+
+            # A file can contain one config context dict or a list thereof
```

Comment:
This assertion seems to only be related to the way you've defined the yaml, or is this an explicit design choice to support lists of objects as well? If so, it should be deliberately documented with included examples for both JSON and YAML.